SITL: Added setup scripts for Windows (Cygwin) users

stephendade opened the pull request:
Added scripts for Windows (Cygwin) users that automatically downloads and installs the SITL environment. There are also scripts for running APM SITL and updating the APM source.

Full documentation at https://github.com/stephendade/SimpleSITL

khancyr commented on Tools/autotest/win_sitl/RunCopter.bat:
```diff
@@ -0,0 +1,4 @@
+rem File run APM:Copter SITL
+rem Assumes a Cgywin install at C:\cygwin
+chdir C:\cygwin\bin
+bash --login -i -c "cd ./ardupilot/ArduCopter && ../Tools/autotest/sim_vehicle.py -j4"
```
don't  need the -j4 , Waf will take care of that !

khancyr commented on Tools/autotest/win_sitl/RunPlane.bat:
```diff
@@ -0,0 +1,4 @@
+rem File run APM:Plane SITL
+rem Assumes a Cgywin install at C:\cygwin
+chdir C:\cygwin\bin
+bash --login -i -c "cd ./ardupilot/ArduPlane && ../Tools/autotest/sim_vehicle.py -j4"
```
don't  need the -j4 , Waf will take care of that !

khancyr commented:
Can you add your readme too ?

khancyr commented on Tools/autotest/win_sitl/InstallCygwinAPM.ps1:
```diff
@@ -0,0 +1,27 @@
+Start-Process -wait -FilePath $PSScriptRoot\setup-x86_64.exe -ArgumentList "--root=C:\cygwin --no-startmenu --local-package-dir=$env:USERPROFILE\Downloads --site=http://cygwin.mirror.constant.com --packages autoconf,automake,ccache,gcc-g++,git,libtool,make,gawk,libexpat-devel, libxml2-devel,python-libxml2,libxslt-devel,python-devel,procps-ng --quiet-mode"
+
+Write-Output "Configuring Cygwin (4/6)"
+Copy-Item apm_install.sh C:\cygwin\home
```
apm_install.sh is missing

khancyr replied:
and it shouldn't be copied in C:\cygwin\home\username ?

stephendade replied:
Can't copy into C:\cygwin\home\username, as we don't know the user's name. Instead we copy into C:\cygwin\home and then in Cygwin, run ..\apm_install.sh. It will then git clone into C:\cygwin\home\username

khancyr commented:
I don't know if it is possible, but to use this files we need to git clone ardupilot and then the repo is clone again in cygwin directory . Could it be possible to point on the previously cloned repo ? Otherwise I think user will be confused .

khancyr commented:
Ok ! It is working after adding apm_install.sh ! Nice !

stephendade commented:
I've added the missing apm_install.sh, readme and removed the -j4's from the run scripts.

> I don't know if it is possible, but to use this files we need to git clone ardupilot and then the repo is clone again in cygwin directory . Could it be possible to point on the previously cloned repo ? Otherwise I think user will be confused .

The idea is more that the scripts are stored in an easy-to-patch-or-fix place. Perhaps the SITL wiki can point the user to these particular files to download and run?

tridge commented:
thanks Stephen!
